libs/platform/src/lib/components/table/table.component.ts

```diff
@@ -211,60 +382,374 @@ export class TableComponent implements OnChanges, OnInit, AfterViewInit, OnDestr
         }
 
         this._reset();
-        const alreadySelected = this._rows.find(r => r.checked === true);
+
+        const alreadySelected = this._rows.find((r) => r.checked === true);
+
         if (alreadySelected) {
             alreadySelected.checked = false;
         }
 
         if (alreadySelected === row) {
             this._unchecked.push(alreadySelected.value);
-            this._emitChange(index);
+            this._emitSelectionChange(index);
 
             return;
         }
 
         this._rows[index].checked = true;
         this._checked.push(row.value);
-        this._emitChange(index);
+        this._emitSelectionChange(index);
     }
 
-    /** @hidden Select/unselect all rows in 'multiple' mode. */
+    /** @hidden Select/Unselect all rows in 'multiple' mode. */
     selectAll(checked: boolean): void {
         this._reset();
-        this.checkedAll = checked;
+        this._checkedAll = checked;
 
         if (checked) {
             this._checkAll();
-            this._emitChange();
+            this._emitSelectionChange();
 
             return;
         }
 
         this._uncheckAll();
-        this._emitChange();
-    }
-
-    /** @hidden */
-    getCellValue(key: string, row: SelectableRow): any {
-        return key.split('.').reduce((a, b) => a[b], row.value);
+        this._emitSelectionChange();
     }
 
     /** @hidden */
     onKeydown(event: KeyboardEvent): void {
         // TODO: since the table should be able the focusable cells, needs to implement arrow buttons navigation in next phases
 
         event.stopPropagation();
-        const click = new MouseEvent('click');
 
         if (KeyUtil.isKeyCode(event, [SPACE, ENTER])) {
+            const click = new MouseEvent('click');
             event.target.dispatchEvent(click);
             event.preventDefault();
         }
     }
 
+    /** @hidden */
+    sort(field: string, direction: SortDirection): void {
+        this._tableService.sort(field, direction);
+        this._popoverOpen = false;
+    }
+
+    /** @hidden */
+    filter(field: string, value: string): void {
+        this._tableService.filter([
+            {
+                field: field,
+                value: value,
+                strategy: CollectionStringFilterStrategy.CONTAINS
+            } as CollectionFilter
+        ]);
+
+        this._popoverOpen = false;
+    }
+
+    /** @hidden */
+    group(field: string): void {
```

Comment:
Is Multi column groups in other Table X phase ?

Reply:
Yes, it is